### ISSUE TITLE
Absent Maintainer?

### ISSUE THREAD

**Issue by hdon** [msmhq/msm#381]
I just saw a very simple PR from 2014 that is still open on this project.

The last commit was from last year.

Does this project need assistance?

Is it dead?

Are people using alternatives?

Thanks!

**Comment by hdon** [comment]
It also looks like CI is down

**Comment by endofcake** [comment]
It has not been maintained for several years. I've got admin rights passed to me some time ago and am trying to keep the project on live support and merge simple PRs, but I'm not planning to actively develop new functionality. Never had time to go and clean up the old issues (see also https://github.com/msmhq/msm/issues/356).

**Comment by hdon** [comment]
Might be worth it to mention this in the README

**Comment by hdon** [comment]
@endofcake have you considered writing a post-mortem for the project? Maybe a description of the some of the challenges `msm` tackled or didn't tackle, what works well and what is hard to maintain, etc?

**Comment by endofcake** [comment]
That's an interesting idea @hdon :) Never really thought about it, as I can't say I know the project very well, I am just a user with some merge rights.

**Comment by Prizmagnetic** [comment]
At the very least, please say that the project is dead right in the README. That way, somebody looking for a currently working solution doesn't waste their time.

**Comment by kjkazinski** [comment]
Prizmagnetic,
There are people who are updating the scripts.


    On Tuesday, January 7, 2020, 11:13:37 PM CST, Prizmagnetic <notifications@github.com> wrote:  
 
 
At the very least, please say that the project is dead right in the README. That way, somebody looking for a currently working solution doesn't waste their time.

**Comment by lemoney** [comment]
@hdon @endofcake I'd like to volunteer to help maintain and work the PR backlog. I personally use this tool often (long time lurker) and with the world becoming more remote it would be nice to be able to help this tool out as people like me continue to spin up servers for friends/family/company/etc.

**Comment by endofcake** [comment]
Thank you for your offer, @lemoney .
I just checked and can see that I don't have admin rights (can't recall if I even had them now) - I can merge PRs, but can't add new maintainers to the project.
@marcuswhybrow, would you be able to assist us here? There are people that are still using the project and would like to salvage it.

**Comment by endofcake** [comment]
In any case, do you see any PRs that should be merged? I think [this one](https://github.com/msmhq/msm/pull/391/files) is the most recent, do you have capacity to verify that it works as expected?

**Comment by lemoney** [comment]
Hey I wanted to run this on one of my servers before giving a thumbs up (also had to fix a filter in my notification emails, sorry!) It does work for my machine

**Comment by M-D-M** [comment]
@endofcake If I remember correctly from when I was added, I think Marcus said if you have a PR request approved, you're automatically added as a maintainer?  Does that sound right?

**Comment by endofcake** [comment]
@M-D-M , yes I think so, that's how I got merge permissions. However I can't add other people as maintainers, and can't fix the CI system which isn't working anymore.

**Comment by mikehebert** [comment]
@endofcake Maybe you should set up a new repository and update the README on this one to point to the new one?

**Comment by endofcake** [comment]
Might be the way to go @mikehebert , but would require considerable effort to get the project up to its former glory. Would prefer not to create a fork if we can avoid it at all.

**Comment by dmgm** [comment]
I'm using these scripts, but if the original owner has abandoned the project I'd think a new fork would be the way to go, if just so that github can operate properly.

I've just installed a bedrock minecraft server on my home linux server, and I did that with annother set of scripts someone wrote and published to github. Both systems rely on screen to run in the background. I'd like to stick with one system if possible. I'd probably be prepared to try and make this work, but if you can't accept patches, then the project really is dead.

**Comment by M-D-M** [comment]
Yeah -- I'm going to try emailing Marcus directly to see if he can add another admin or two to the project.  I'll update here in a week if I get a response.

**Comment by lemoney** [comment]
Yay! That'd be excellent. This is still a great tool

**Comment by renderorange** [comment]
I used to have write permissions to merge pull requests on the repo but I don't seem to have that anymore.  I'm not sure if it was something I changed not, it's been a long time since I've done any work here.

@zachlatta might also have admin on the project to add other admins.  He was added as maintainer after Marcus.

**Comment by zachlatta** [comment]
Hey there, I think I still have admin. When I get back to my computer next week, I’ll do my best to check in on this and add someone else as an admin so the project doesn’t die

**Comment by M-D-M** [comment]
@zachlatta Any luck with adding someone else as an admin?

I apologize to everyone for my late reply! Like everyone else, my year has been busy just trying to survive. :(

The response I got from @marcuswhybrow when I emailed him was:

> I'd be very happy to add new maintainers to the project.
I see that you are both already "Collaborators" and therefore have write access and the ability to merge pull requests.
Is there some additional action you need me to take?

**Comment by zachlatta** [comment]
Wow, thanks so much for the ping! I just added you as an admin to the project @M-D-M. @marcuswhybrow: I hope that was OK!

**Comment by M-D-M** [comment]
@zachlatta That appears to have worked!  Thanks!

@renderorange and @lemoney Going to get you added as maintainers of some sort.  Will post back here once done.

**Comment by renderorange** [comment]
> @renderorange and @lemoney Going to get you added as maintainers of some sort. Will post back here once done.

@M-D-M I've created new releases and tested and merged pull requests in the past; I'm happy to continue helping with that if need, depending on how active others are wanting to be involved.  I don't want to steal the opportunity for others to contribute if people are looking to help with this project.

We also need to get the automated testing working.  I'm happy to get that working again too, then expanding what's actually tested through it.

@endofcake has been merging pull requests and making releases for a lot of years since I've been active.  I can't speak for them and their availability, but should perhaps loop them in too if they want to still be involved.

**Comment by renderorange** [comment]
@M-D-M thanks!  I still see "you need write permissions" on pull requests for this project.  Not sure if that's part of the maintainer priv or not.

**Comment by renderorange** [comment]
@M-D-M I've had a little more time lately in the evenings to help with development and pull requests.  I think I'm still missing maintainer privs, though; I don't see the ability to do anything with pull requests on the project.
